### DISCUSSION TITLE
Please stop with all the individual request regarding pre-compiled firmwares

### DISCUSSION THREAD

**Discussion by Schnabulation** [Jyers/Marlin#94] [Q&A] [answered]
Hello guys

In the last days I have noticed a very high spike in personal requests regarding pre-compiled firmwares. Requests like "can you enabled this in your pre-compiled firmware" or "can you compile your firmware for that specific use case".

While I understand that a lot of you might find it difficult to compile your own firmware based on this fork I personally find that Jyers should focus his time into improving the code itself and not be your personal firmware modifier and compiler. Only because Jyers is easier to reach and convince than the folks over at the official Marlin repo doesn't mean that you should do that.

Marlin is open source and if you want your personal firmware with your own configuration then please spend the time and learn how to compile it yourself. It is litterally 30 minutes of your time and then you can do it on your own.

Thank you.

**Comment by fakezeta** [comment]
Hi Schnabulation,

I second your view even if I made a similar request and really takes even less than 30 minute to compile our own build.
Even though a poll with the most common configuration option could be useful to reach a broader population.

For example I proposed an Inverted E build that could be user by who has a geared extruder, if it's a common required option could be worth including in the precompiled build otherwise I will continue to build it on my own :) 

Cheers

**Reply by t3chguy** [reply]
For Inverted E I really think Marlin should support negative axis step values, it'd mean it can be done via GCode.
The issue with adding such a build is that it won't be one build, it'd be one for each bed mesh size, etc, then also double everything because one for v4 and one for v427

**Reply by fakezeta** [reply]
Yep, quite too many builds. I discovered that Marlin does not support negative number by trial and error ^_^

**Reply by t3chguy** [reply]
Same, seems like such a simple solution to the problem

**Reply by amandamana** [reply]
Both seem like strange workarounds instead of properly wiring the extruder. Does that have some disadvantage?

**Reply by Woodcat64** [reply]
It's probably faster to swap two wires, than compile new fw anyway.

**Reply by t3chguy** [reply]
It also tends to be saner from a hardware standpoint to have all your steppers wired up the same in case you need to swap/extend wiring

**Comment by Jyers** [comment, marked as answer]
Thanks for this!

**Comment by rorywquin** [comment]
To me a branch is something in a tree and a fork is what I eat with. I cannot 'git' my head around how GitHub works. I did compile my own FW when I first started but soon realised that the work like what Jyers does means that I can simply use precompiled FW 99% of the time. HOWEVER, I would like to be able to get (for example) the '4.2.2 5x5 fast' source code and take a look around & possibly do my own tweaks. I suspect that it is available but I just don't know how to get to it! Any chance of some pointers?

**Reply by Jyers** [reply]
If you look in the configuration files folder there's a folder called e3v2 templates that has a few configurations for your average setup. You can grab one of those configurations and make any changes you need from there.

**Reply by Woodcat64** [reply]
For the 5x5 grid, search in Configuration.h for GRID_MAX_POINTS_X and change the value to 5. To see what was changed, you can download the generic Configuration.h from the buxfix.2.0 branch and compare the files.
